arbnode/seq_coordinator.go

```diff
@@ -573,6 +580,15 @@ func (c *SeqCoordinator) deleteFinalizedMsgsFromRedis(ctx context.Context, final
 	return nil
 }
 
+func (c *SeqCoordinator) blockMetadataAt(ctx context.Context, pos arbutil.MessageIndex) arbostypes.BlockMetadata {
+	blockMetadataStr, err := c.Client.Get(ctx, redisutil.BlockMetadataKeyFor(pos)).Result()
+	if err != nil {
+		log.Debug("SeqCoordinator couldn't read blockMetadata from redis", "err", err, "pos", pos)
+		return nil
+	}
```

Comment:
This should distinguish missing data, which I think is legitimate since we might not have metadata, from another error from the redis client like missing data.
Something like:
```
   if err != nil {
        if errors.Is(err, redis.Nil) {
            // Legitimate case of no metadata
            return nil, nil
        }
        // Real error reading from Redis
        return nil, fmt.Errorf("failed reading block metadata: %w", err)
    }
```

Then in update() you can add error handling matching the existing error flow for messages:
```
    metadata, err := c.blockMetadataAt(ctx, msgToRead)
    if err != nil {
        log.Warn("coordinator failed reading block metadata", "pos", msgToRead, "err", err)
        msgReadErr = err // Use existing error flow
        break
    }
    blockMetadataArr = append(blockMetadataArr, metadata)
    msgToRead++
```

This will ultimately cause a retry
```

	if (wantsLockoutErr != nil) || (msgReadErr != nil) {
		return c.retryAfterRedisError()
	}
```